File: 151-reverse-words/Solution.java
import java.util.*;

class Solution151 {
    public String P0151_reverseWords(String s) {
        if (s == null) return null;
        String[] splitString = s.trim().split("\\s+");
        List<String> reversed_List = new ArrayList<>(Arrays.asList(splitString));
        return String.join(" ", reversed_List.reversed());
    }

    static void main() {
        Solution151 sol = new Solution151();
        System.out.println(sol.P0151_reverseWords("   a  good   example   "));
    }
}

/*
public static String P0151_reverseWords(String s) {
    if (s == null) return null;
    String[] parts = s.trim().split("\\s+");
    java.util.List<String> reversed_List = new java.util.ArrayList<>(java.util.Arrays.asList(parts));
    java.util.Collections.reverse(reversed_List);
    String result = String.join(" ", reversed_List);
    return result;
}

public static void main(String[] args) {
    System.out.println(P0151_reverseWords("a good   example "));
}

    for (String s1 : reversed_List){
        System.out.println(s1);
        if (s1.isEmpty()) {
            reversed_List.remove(s1);
        }
    }

String result = String.join(" ", reversed_List);
*/

File: 151-reverse-words/solution.py
class Solution1122:
    def P0151_reverseWords(self, s: str) -> str:
        list1 = s.split(' ')
        print(type(list1))

        while '' in list1:#好理解但效率差
            list1.remove('')
        #最優寫法clean_list = [item for item in my_list if item != '']
        list1.reverse()
        print(list1)
        result = ' '.join(list1)
        return result

sol = Solution1122()
print(sol.P0151_reverseWords("  hello world  "))

"""
你有什麼想法?
strip??
split??
先存成List??再顛倒?
首先要得到一個乾淨的列表
現在我們得到一張乾淨列表了
你有什麼想法?
把列表整個顛倒過來
"""